### PR TITLE
Reduce size of CUDA tests

### LINKED ISSUE #245: Reduce size of CUDA tests
Currently, our CUDA tests are rather memory hungry. This is fine for running the test suites using A100 GPUs.

However, we don't need "all that memory" for the purposes of running tests. I have local system with a 1 GB NVIDIA GPU, and the CUDA tests take almost 17 minutes to finish.

So, we should work on paring down the runtimes and/ or sizes of the tests.

### PR DESCRIPTION
**Pull request by lukem12345**
Close #245

### CODE REVIEW

**Comment by lukem12345** [comment]
The changes in 6810ee6 brought down the time for the CUDA testset from 16m:43s to 4m:29s .

File: test/cuda_sims.jl [hunk 1]
```diff
@@ -189,7 +189,7 @@ end
          a = 0.94,
          ν = 182.5)
 
-  tₑ = 10.0
+  tₑ = 1.0
```
Comment (GeorgeR227) [inline test/cuda_sims.jl:192]: Is this enough time to see the formation and stabilization of at least one ridge?

File: test/cuda_sims.jl [hunk 2]
```diff
@@ -274,7 +274,7 @@ end
          a = 0.94f0,
          ν = 182.5f0)
 
-  tₑ = 3.0f0
+  tₑ = 1.0f0
```
Comment (GeorgeR227) [inline test/cuda_sims.jl:277]: Same as before, is this enough to see at least a single ridge form and stabilize?

File: test/cuda_sims.jl [hunk 3]
```diff
@@ -148,7 +148,7 @@ end
     F = t -> t ≥ 1.1 ? F₂ : F₁)
 
   prob = ODEProblem(cpu_fₘ, cpu_u₀, (0, tₑ), cpu_constants_and_parameters)
-  cpu_soln = solve(prob, Tsit5())
+  cpu_soln = solve(prob, Tsit5(), save_everystep=false, save_idxs=[:U])
```
Comment (GeorgeR227) [inline test/cuda_sims.jl:151]: Don't need this `save_idxs` since `:U` is the only thing we can track.

**Review by GeorgeR227 — CHANGES_REQUESTED** [review]

File: test/cuda_sims.jl (hunk 3 above)
Comment (lukem12345) [reply test/cuda_sims.jl:151]: Can't :V also be tracked?

Comment (GeorgeR227) [reply test/cuda_sims.jl:151]: Yeah you're right, I thought this was for Heat.

File: test/cuda_sims.jl (hunk 1 above)
Comment (lukem12345) [reply test/cuda_sims.jl:192]: A bump forms at 2.5 so I'll crank it back up.

**Comment by lukem12345** [comment]
Upping the time of the Klausmeier sims puts time for my system to 5m:53s

**Pull request merged**